go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -271,29 +274,68 @@ func (c *Controller) syncServices(key string) error {
 
 	vips := collectServiceVIPs(service)
 	if svcNeedsIdling(service.Annotations) && !util.HasValidEndpoint(service, endpointSlices) {
+		// addServiceToIdlingBalancer adds the vips to service tracker
 		err = c.addServiceToIdlingBalancer(vips, service)
 		if err != nil {
 			c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToAddToIdlingBalancer",
 				"Error trying to add to Idling LoadBalancer for Service %s/%s: %v", name, namespace, err)
 			return err
 		}
-		vipsTracked = vipsTracked.Delete(vips.List()...)
-		err = deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)
+		// create a temporary map to convert sets.String to map for passing to functions
+		tmpVIPs := make(map[string]string)
+		for v := range vips {
+			delete(vipsTracked, v)
+			tmpVIPs[v] = ""
+		}
+		needsCleanUp := false
+		for vipProtocol := range vips {
+			vip, _ := splitVirtualIPKey(vipProtocol)
+			if c.serviceTracker.getLoadBalancer(name, namespace, vip) != loadbalancer.IdlingLoadBalancer {
```

Comment:
Can we merge the two range loops and add to tmpVIPs only if this if matches?
So tmpVIPs contains only those vip that need to be removed from non idling loadbalancers.
With that, you can get rid of needsCleanUp because the map will be empty if there are no matches for the if

Reply:
good idea, will do